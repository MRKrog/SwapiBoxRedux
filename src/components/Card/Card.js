import React from 'react';
import './Card.scss';
import PropTypes from 'prop-types';

const Card = ({ info, handleFavBtn }) => {
  return (
    <div className="Card">
      <section className="Card-Title">
        <h4>{info.name}</h4>
        <button className={info.favorite ? 'activeFav' : 'inactiveFav'}
                onClick={() => handleFavBtn(info.name, info.category)}>
          <i className="fas fa-star"></i>
        </button>
      </section>
      <section className="Card-Body">
        {info.homeworld && <p>Homeworld: <span>{info.homeworld}</span></p>}
        {info.species && <p>Species: <span>{info.species}</span></p>}
        {info.language && <p>Language: <span>{info.language}</span></p>}
        {info.population && <p>Population: <span>{info.population}</span></p>}
        {info.model && <p>Model: <span>{info.model}</span></p>}
        {info.class && <p>Class: <span>{info.class}</span></p>}
        {info.passenger && <p>Passengers: <span>{info.passenger}</span></p>}
        {info.climate && <p>Climate: <span>{info.climate}</span></p>}
        {info.terrain && <p>Species: <span>{info.terrain}</span></p>}
        {info.residents && <ul>Residents: {info.residents.map((val, i) => <li key={i}>{val}</li>) }</ul>}
      </section>
    </div>
  )
}

Card.propTypes = {
  info: PropTypes.object,
  handleFavBtn: PropTypes.func
}

export default Card;
