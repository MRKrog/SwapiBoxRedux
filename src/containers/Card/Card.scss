$font-title: 'Abel', sans-serif;
$font-copy: 'Lato', sans-serif;
$font-Label: 'Rajdhani', sans-serif;
// Rajdhani
// Barlow


.Card {
  margin: 1%;
  width: 23%;
  display: flex;
  flex-direction: column;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  box-shadow: 1px 2px 8px 0px #000000a6;
  background: #404040;

  @media only screen and (max-width: 1060px) {
    width: 31%;
  }

  @media only screen and (max-width: 960px) {
    width: 48%;
  }

  .Card-Title {
    display: flex;
    align-items: center;
    justify-content: space-between;
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
    background: #353031;
    padding: 0 .75em;
    line-height: 2em;

    h4 {
      font-family: $font-Label;
      font-weight: bold;
      font-size: 1.2em;
      margin: 0;
      letter-spacing: .05em;
      color: #6d6d6d;
    }

    button {
      background: none;
      border: none;
      outline: none;
      font-size: 1em;
      cursor: pointer;
      background: #8a7c50;
      padding: 4px;
      border-radius: 5px;
      line-height: 0;
      transition: all .5s ease;

      &:hover {
        color: yellow;
      }

      i {
        font-size: .75em
      }
    }

    .activeFav {
      color: yellow;
    }
    .inactiveFav {
      color: white;

    }
  }


  .Card-Body {
    padding: .5em .75em;

    p {
      display: flex;
      flex-direction: column;
      font-size: 1em;
      letter-spacing: .05em;
      margin: 0;
      color:#fff;
      font-family: $font-Label;

      span {
        color: #a7a7a7;
        line-height: 1.3;
        font-weight: 100;
        font-size: 1em;
        font-family: $font-title;
        padding-bottom: .5em;
        padding-top: .2em;
      }
    }
    ul {
      padding:0;
      margin:0;
      font-size: 1em;
      letter-spacing: .05em;
      margin: 0;
      color: #fff;
      font-family: $font-Label;

      li {
        list-style-type: none;
        display: flex;
        align-items: center;
        color: #a7a7a7;
        line-height: 1.3;
        font-weight: 100;
        font-size: .9em;
        font-family: $font-copy;
      }
    }

    h5 {
      margin:0;

      span {
        margin:0;
      }
    }
  }
}
