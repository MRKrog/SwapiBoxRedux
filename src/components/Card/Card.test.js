import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

const mockHandleFavBtn = jest.fn()

const info = {
      name: 'Luke Skywalker',
      category: "people",
      favorite: false,
      homeworld: "Tatooine",
      language: "Galactic Basic",
      name: "Luke Skywalker",
      population: "200000",
      species: "Human"
    }

describe('Card', () => {
  let wrapper;

  beforeEach(() => {
      wrapper = shallow(<Card info={info}
                              handleFavBtn={mockHandleFavBtn}
                              />)

  })

  it("should match snapshot when all data is passed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke handleFavBtn when button clicked and called with two name and category', () => {
    let button = wrapper.find('button')
    button.simulate('click')
    expect(mockHandleFavBtn).toHaveBeenCalledWith('Luke Skywalker', 'people')
  })

  it("should change its class to activeFav when favorite is true", () => {
    const newInfo = {
          name: 'Luke Skywalker',
          category: "people",
          favorite: true,
          homeworld: "Tatooine",
          language: "Galactic Basic",
          name: "Luke Skywalker",
          population: "200000",
          species: "Human"
        }

    let newWrapper = shallow(<Card info={newInfo} />)

    expect(newWrapper).toMatchSnapshot();
  });

})
