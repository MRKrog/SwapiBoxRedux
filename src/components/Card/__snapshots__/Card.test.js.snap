// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should change its class to activeFav when favorite is true 1`] = `
<div
  className="Card"
>
  <section
    className="Card-Title"
  >
    <h4>
      Luke Skywalker
    </h4>
    <button
      className="activeFav"
      onClick={[Function]}
    >
      <i
        className="fas fa-star"
      />
    </button>
  </section>
  <section
    className="Card-Body"
  >
    <p>
      Homeworld: 
      <span>
        Tatooine
      </span>
    </p>
    <p>
      Species: 
      <span>
        Human
      </span>
    </p>
    <p>
      Language: 
      <span>
        Galactic Basic
      </span>
    </p>
    <p>
      Population: 
      <span>
        200000
      </span>
    </p>
  </section>
</div>
`;

exports[`Card should match snapshot when all data is passed correctly 1`] = `
<div
  className="Card"
>
  <section
    className="Card-Title"
  >
    <h4>
      Luke Skywalker
    </h4>
    <button
      className="inactiveFav"
      onClick={[Function]}
    >
      <i
        className="fas fa-star"
      />
    </button>
  </section>
  <section
    className="Card-Body"
  >
    <p>
      Homeworld: 
      <span>
        Tatooine
      </span>
    </p>
    <p>
      Species: 
      <span>
        Human
      </span>
    </p>
    <p>
      Language: 
      <span>
        Galactic Basic
      </span>
    </p>
    <p>
      Population: 
      <span>
        200000
      </span>
    </p>
  </section>
</div>
`;
